refactor(daffio): replace deprecated structural directives with control flow syntax

JeetuSuthar opened the pull request:
issue : #4057

JeetuSuthar committed:
refactor(daffio): replace deprecated structural directives with control flow syntax

Replace *ngIf/*ngFor uses with @if/@for in daffio templates:
- footer/docs-footer component
- footer/footer component
- nav/sidebar-body component
- docs/packages/package-cards component

Closes #4057

xelaint reviewed (changes requested):
Please check the build log as the updates you made are incorrect: https://github.com/graycoreio/daffodil/actions/runs/18173432520/job/51733417047?pr=4061

damienwebdev commented on apps/daffio/src/app/core/footer/docs-footer/docs-footer.component.html:
```diff
@@ -1,20 +1,20 @@
 <daff-container size="lg">
   <div class="daffio-docs-footer__grid">
     <div class="daffio-docs-footer__links">
```
This file is already correct and doesn't need modifications.

damienwebdev commented on apps/daffio/src/app/core/nav/sidebar-body/component.html:
```diff
@@ -1,11 +1,11 @@
 <daff-nav-list>
-  @for (link of links$ | async; track $index) {
+    @for (let link of links$ | async) {
```
This file is already correct and doesn't need modifications.

xelaint replied:
@JeetuSuthar The changes you made to this file is still not necessary. The originally implementation was correct.

JeetuSuthar committed:
fix(daffio): correct footer component @for syntax and indentation

- Fixed misplaced Sponsor link that was outside @for closing brace
- Ensured proper indentation and syntax structure
- Addresses build syntax errors reported in PR review

damienwebdev commented:
@JeetuSuthar The code in several of your PRs represents limited understanding of the underlying syntax. While I appreciate the PRs, I'm going to close these. If you'd like to resubmit, please build your PRs locally and ensure the build passes before submitting again.

If you don't know how to build the library, we have instructions in the [contribution guidelines](https://github.com/graycoreio/daffodil/blob/develop/CONTRIBUTING.md),
